File: hrm/hrm/report/bank_salary_register/bank_salary_register.js
// Copyright (c) 2016, AVU and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Bank Salary Register"] = {
	"filters": [
        {
			"fieldname":"payroll_period",
			"label": __("Payroll Period"),
			"fieldtype": "Link",
			"options": "Payroll Period",
			"reqd": 1
		},
		{
			"fieldname":"employee",
			"label": __("Employee"),
			"fieldtype": "Link",
			"options": "Employee"
		},
		{
			"fieldname":"company",
			"label": __("Company"),
			"fieldtype": "Link",
			"options": "Company",
			"default": frappe.defaults.get_user_default("Company")
		},
		{
			"fieldname":"docstatus",
			"label":__("Document Status"),
			"fieldtype":"Select",
			"options":["Draft", "Submitted", "Cancelled"],
			"default":"Submitted"
		}

	]
    ,
	"formatter":function(value, row, column, data, default_formatter) {
		var stop_loop = setInterval(()=>{
			$(".datatable").attr("dir", "rtl")
			clearInterval(stop_loop)
		})
		value = default_formatter(value, row, column, data);
		
		return value;
	}

};

File: hrm/hrm/report/bank_salary_register/bank_salary_register.py
# Copyright (c) 2013, AVU and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe.utils import flt
import json
from frappe import _
from stira.stira.report.utils import get_columns, get_values_list

def execute(filters=None):
	from_date, to_date = frappe.db.get_value("Payroll Period", filters.get("payroll_period"), ["start_date", "end_date"]) if filters.get("payroll_period") else ['', '']
	filters.update({"from_date": from_date, "to_date": to_date})
	frappe.errprint(str(filters))
	doc_status = {"Draft": 0, "Submitted": 1, "Cancelled": 2}
	# if not filters: filters = {}
	sql="call BankSalaryRegister({0},{1},{2},'{3}','{4}')".format(json.dumps(filters.get('company')),json.dumps(filters.get('employee')),json.dumps(doc_status[filters.get("docstatus")]),filters.get('from_date'),filters.get('to_date'))
	frappe.errprint(str(sql))
	dict = frappe.db.sql(sql, as_dict=1)
	columns = get_columns(dict)
	data = get_values_list(dict)
	return columns, data

def get_conditions(filters):
	conditions = ""
	doc_status = {"Draft": 0, "Submitted": 1, "Cancelled": 2}

	if filters.get("docstatus"):
		conditions += "docstatus = {0}".format(doc_status[filters.get("docstatus")])

	if filters.get("from_date"): conditions += " and start_date >= %(from_date)s"
	if filters.get("to_date"): conditions += " and end_date <= %(to_date)s"
	if filters.get("company"): conditions += " and company = %(company)s"
	if filters.get("employee"): conditions += " and employee = %(employee)s"

	return conditions, filters

